examples/tree_2d_dgsem/elixir_euler_colliding_flow.jl

```diff
@@ -0,0 +1,110 @@
+
+# shock capturing necessary for this tough example, however alpha_max = 0.5 is fine
+indicator_sc = IndicatorHennemannGassner(equations, basis,
+                                         alpha_max=0.5, 
+                                         alpha_min=0.0001,
+                                         alpha_smooth=true,
+                                         variable=density_pressure)
+volume_integral = VolumeIntegralShockCapturingHG(indicator_sc;
+                                                 volume_flux_dg=volume_flux,
+                                                 volume_flux_fv=surface_flux)
+solver = DGSEM(basis, surface_flux, volume_integral)
+
+coordinates_min = (-64.0, -64.0)
+coordinates_max = ( 64.0,  64.0)
+
+# only refinment in a patch. Needs x=-17/+17 to trigger refinment due to coarse base mesh
+refinement_patches = (
+  (type="box", coordinates_min=(-17, -64), coordinates_max=(17, 64)),
+  (type="box", coordinates_min=(-17, -64), coordinates_max=(17, 64)),
+  (type="box", coordinates_min=(-17, -64), coordinates_max=(17, 64)),
+  (type="box", coordinates_min=(-17, -64), coordinates_max=(17, 64)),
+  #(type="box", coordinates_min=(-17, -64), coordinates_max=(17, 64)), # very high resolution, takes about 1000s on 2 cores
+)
+mesh = TreeMesh(coordinates_min, coordinates_max,
+                initial_refinement_level=3,
+                refinement_patches=refinement_patches,
+                periodicity=(false,true),
+                n_cells_max=100_000)
+semi = SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver, boundary_conditions=boundary_conditions)
+
+###############################################################################
+# ODE solvers, callbacks etc.
+
+tspan = (0.0, 25.0)
+ode = semidiscretize(semi, tspan)
+
+summary_callback = SummaryCallback()
+
+analysis_interval = 1000
+analysis_callback = AnalysisCallback(semi, interval=analysis_interval)
+
+alive_callback = AliveCallback(analysis_interval=analysis_interval)
+
+save_solution = SaveSolutionCallback(interval=1000,
+                                     save_initial_solution=true,
+                                     save_final_solution=true,
+                                     solution_variables=cons2prim)
+
+callbacks = CallbackSet(summary_callback,
+                        analysis_callback, alive_callback, 
+                        save_solution)
+
+# positivity limiter necessary for this tough example
+limiter! = PositivityPreservingLimiterZhangShu(thresholds=(5.0e-6, 5.0e-6),
+                                               variables=(Trixi.density, pressure))
+stage_limiter! = limiter!
+step_limiter!  = limiter!
+
+###############################################################################
+# run the simulation
+# use adaptive time stepping based on error estimates, time step roughly dt = 5e-3
+sol = solve(ode, SSPRK43(stage_limiter!, step_limiter!),
+            save_everystep=false, callback=callbacks);
```

Comment:
```suggestion
stage_limiter! = PositivityPreservingLimiterZhangShu(thresholds=(5.0e-6, 5.0e-6),
                                                     variables=(Trixi.density, pressure))

###############################################################################
# run the simulation
# use adaptive time stepping based on error estimates, time step roughly dt = 5e-3
sol = solve(ode, SSPRK43(stage_limiter!),
            save_everystep=false, callback=callbacks);
```
The `step_limiter!` is not necessary if you have a `stage_limiter!`, see https://github.com/SciML/OrdinaryDiffEq.jl/blob/cdc5ef291d37f31ce7746e9f27aeeeae5d806393/src/perform_step/ssprk_perform_step.jl#L1031-L1032